src/karmen_frontend/src/components/printers/webcam-stream.js

```diff
@@ -63,22 +63,53 @@ export const WebcamStreamRenderer = ({
   if (rotate90) {
     klass.push("rotate-90");
   }
-  const isStreamAvailable = image && printerUtils.isConnected(printer);
-  return (
-    <>
-      <div
-        className={`webcam-stream ${isStreamAvailable ? "" : "unavailable"}`}
-      >
-        {isStreamAvailable ? (
+
+  let image, status_code;
+  if (imageResponse) {
+    image = imageResponse[0];
+    status_code = imageResponse[1];
+  }
+  // No response but printer online -> fresh page load
+  // No response and printer is offline -> stream unavailable
+  // 404 -> stream unavailable
```

Comment:
The server should never return 404 in case the stream is not available.

Reply:
Shouldn't, but it does: https://github.com/fragaria/karmen/blob/master/src/karmen_backend/server/routes/printers.py#L497